File: Elitesportweb/models/Payment.js
import mongoose from 'mongoose'

const PaymentSchema = new mongoose.Schema({
  memberId: { type: String, required: true },
  memberName: { type: String, required: true },
  classId: { type: String, required: true },
  className: { type: String, required: true },
  paymentType: { type: String, required: true, enum: ['admission', 'monthly', 'sixMonthly', 'annually'] },
  amount: { type: Number, required: true },
  paymentMonth: { type: String, required: true }, // YYYY-MM format
  paymentDate: { type: Date, default: Date.now },
  status: { type: String, default: 'paid', enum: ['paid', 'pending', 'overdue'] },
  paymentMethod: { type: String, enum: ['cash', 'card', 'bank_transfer'] },
  notes: { type: String },
  verificationStatus: { type: String, default: 'unverified', enum: ['unverified', 'requested', 'verified', 'disputed'] },
  verificationRequestDate: { type: Date },
  verifiedBy: { type: String },
  verifiedDate: { type: Date }
})

if (mongoose.models.Payment) {
  delete mongoose.models.Payment
}

export default mongoose.model('Payment', PaymentSchema)

File: Elitesportweb/elite_sports_flutter/lib/models/payment.dart
class Payment {
  final String id;
  final DateTime paymentDate;
  final double amount;
  final String paymentType;
  final String paymentMonth;
  final String className;
  final String? verificationStatus;

  Payment({
    required this.id,
    required this.paymentDate,
    required this.amount,
    required this.paymentType,
    required this.paymentMonth,
    required this.className,
    this.verificationStatus,
  });

  factory Payment.fromJson(Map<String, dynamic> json) {
    return Payment(
      id: json['_id'] ?? '',
      paymentDate: DateTime.tryParse(json['paymentDate'] ?? '') ?? DateTime.now(),
      amount: (json['amount'] ?? 0).toDouble(),
      paymentType: json['paymentType'] ?? '',
      paymentMonth: json['paymentMonth'] ?? '',
      className: json['className'] ?? '',
      verificationStatus: json['verificationStatus'],
    );
  }
}